app/services/moves/importer.rb

```diff
@@ -10,63 +10,72 @@ def initialize(items)
+      new_person_count = 0
+      changed_profile_count = 0
+      people.each do |offender_no, person_data|
+        person = People::Importer.new(person_data).call
+        new_person_count += 1 if person.new_record?
+        person.save!
+        profile = person.latest_profile
+        Alerts::Importer.new(profile: profile, alerts: alerts.fetch(offender_no, [])).call
+        PersonalCareNeeds::Importer.new(profile: profile,
+                                        personal_care_needs: personal_care_needs.fetch(offender_no, [])).call
+        if profile.changed?
+          changed_profile_count += 1
+          profile.save!
+        end
       end
-    end
-
-    def import_personal_care_needs
-      NomisClient::PersonalCareNeeds.get(people_nomis_prison_numbers)
-                                    .group_by { |pcn| pcn.fetch(:offender_no) }
-                                    .each do |offender_no, personal_care_needs_attributes|
-        person = Person.find_by!(nomis_prison_number: offender_no)
-        PersonalCareNeeds::Importer.new(
-          profile: person.latest_profile,
-          personal_care_needs: personal_care_needs_attributes
-        ).call
+      if new_person_count.positive? || changed_profile_count.positive?
+        Rails.logger.info("[Moves::Importer] created #{new_person_count} updated #{changed_profile_count}")
       end
     end
+    # rubocop:enable Style/GuardClause
+    # rubocop:enable Metrics/AbcSize
+    # rubocop:enable Metrics/MethodLength
 
-    def import_move(move)
-      return if update_move_with_same_nomis_event_id(move)
-
-      new_move = Move.new(move_params(move))
-      existing_move = new_move.existing
-
-      if existing_move
-        existing_move.update(move_params(move))
-      else
-        new_move.save
+    # rubocop:disable Metrics/MethodLength
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Style/GuardClause
+    def import_moves
+      new_count = 0
+      update_count = 0
+      items.map { |m| move_params(m) }.each do |move|
+        new_move = Move.new(move)
+        existing_move = Move.find_by_nomis_event_ids([move[:nomis_event_id]]) || new_move.existing
+        if existing_move
+          existing_move.assign_attributes(move)
+          if existing_move.changed?
+            update_count += 1
+            existing_move.save!
+          end
+        else
+          new_count += 1
+          new_move.save!
+        end
+      end
+      if new_count.positive? || update_count.positive?
+        Rails.logger.info("[Moves::Importer] #{new_count} new moves updated #{update_count} moves")
       end
     end
-
-    def update_move_with_same_nomis_event_id(move)
-      same_nomis_event_id_move = Move.find_by_nomis_event_ids([move[:nomis_event_id]])
-      same_nomis_event_id_move&.update(move_params(move))
-    end
+    # rubocop:enable Style/GuardClause
+    # rubocop:enable Metrics/MethodLength
+    # rubocop:enable Metrics/AbcSize
```

Comment:
This refactor seems to introduce two complex, long methods which violate our Rubocop conventions. Could this be submitted in a separate PR to the logging and test changes, so we can demonstrate the tests are successful against the original code?

Reply:
The changes are needed to support the logging. IMHO this new method isn't particularily long or complex (our Rubocop method length is set to the default of 10 which is extremely  short). Open to suggestions.

Reply:
These exceptions all vanish now we have the GDS style guides in place